sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/GcpApiSurfaceTest.java

```diff
@@ -96,7 +107,8 @@ public void testGcpApiSurface() throws Exception {
             classesInPackage("org.apache.beam"),
             classesInPackage("org.apache.commons.logging"),
             classesInPackage("org.codehaus.jackson"),
-            classesInPackage("org.joda.time"));
+            classesInPackage("org.joda.time"),
+            classesInPackage("org.threeten.bp"));
```

Comment:
What is this dependency? Does it need to be on the API surface?

Reply:
This is a dependency of gax-grpc for some Duration classes. It's pulled in by 1.1.0 Spanner release, which now uses a GAX OperationFuture instead of a Spanner Operation in CreateDatabase:

https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseAdminClient.java#L63

We could move it off of the API surface by using a wrapper class instead of exposing the Spanner client objects directly in SpannerAccessor:

https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerAccessor.java

Reply:
Got it. I don't think we need to make significant efforts here to wrap client dependencies. If this becomes problematic (i.e. diamond dependency troubles), we can consider wrapping the client or pushing on the Spanner library to encapsulate the dependency instead.